src/lib/hooks/swap/useWrapCallback.tsx

```diff
@@ -0,0 +1,153 @@
+import { ContractTransaction } from '@ethersproject/contracts'
+import { Trans } from '@lingui/macro'
+import { useWETHContract } from 'hooks/useContract'
+import { atom, useAtom } from 'jotai'
+import { useAtomValue } from 'jotai/utils'
+import { Field, swapAtom } from 'lib/state/swap'
+import tryParseCurrencyAmount from 'lib/utils/tryParseCurrencyAmount'
+import { useCallback, useEffect, useMemo } from 'react'
+
+import { WRAPPED_NATIVE_CURRENCY } from '../../../constants/tokens'
+import useActiveWeb3React from '../useActiveWeb3React'
+import { useCurrencyBalances } from '../useCurrencyBalance'
+import useNativeCurrency from '../useNativeCurrency'
+
+export enum WrapInputError {
+  NO_ERROR, // must be equal to 0 so all other errors are truthy
+  ENTER_NATIVE_AMOUNT,
+  ENTER_WRAPPED_AMOUNT,
+  INSUFFICIENT_NATIVE_BALANCE,
+  INSUFFICIENT_WRAPPED_BALANCE,
+}
+
+export function WrapErrorText({ wrapInputError }: { wrapInputError: WrapInputError }) {
```

Comment:
If this is only used in the toolbar, I think it should live in the Toolbar/Caption file.